engines/sci/engine/klists.cpp

```diff
@@ -824,6 +827,13 @@ reg_t kArraySetElements(EngineState *s, int argc, reg_t *argv) {
 }
 
 reg_t kArrayFree(EngineState *s, int argc, reg_t *argv) {
+	if (s->_segMan->getSegmentType(argv[0].getSegment()) == SEG_TYPE_SCRIPT) {
```

Comment:
In SSCI, this function is named "del", rather than "freeArray" - our naming very confusing, and wrong. This function should only delete a single element from an array, shifting all the rest, and that's what it does, but its naming is wrong. And SSCI does exactly the same for strings, which is what the code that I've included performs. So, it DOES make sense, since it's what the original is doing.... it's our naming of kArrayFree which is confusing.